src/pipelines/master_pipeline.py

```diff
@@ -345,36 +368,46 @@ def get_args(
         if XGB_MAXDEPTH: xgb['x_max_depth'] = XGB_MAXDEPTH
         if XGB_COLSAMPLEBYTREE: xgb['x_colsample_bytree'] = XGB_COLSAMPLEBYTREE
         if XGB_MINCHILDWEIGHT: xgb['x_colsample_bylevel'] = XGB_MINCHILDWEIGHT
+    if SGD:
+        sgd = dict()
+        sgd['alpha'] = SGD_ALPHA
+        sgd['loss'] = SGD_LOSS
+        sgd['penalty'] = SGD_PENALTY
+        sgd['num_epochs'] = SGD_EPOCHS
+        sgd['batch_size'] = SGD_BATCH_SIZE
+        sgd['seed'] = SEED
+        assert HDF5_PATH_TRAIN is not None, "SGD-based methods should be used with HDF5"
 
     algorithms = dict()    
     if log_reg: algorithms['log_reg'] = log_reg
     if svm: algorithms['svm'] = svm
     if xgb: algorithms['xgb'] = xgb
-    #add the memory_net parameters to algorithms as well
     if mem_net:
-        if len(algorithms)>0:
-            print("Error:  The memory network algorithm must be run alone")
-            quit()
         algorithms['mem_net']=mem_net
+    if sgd:
+        algorithms['sgd'] = sgd
+
+    logger.debug("Algorithms structure: {}".format(algorithms))
 
     # Enforce requirement and limitation of one algorithm per run
     if len(algorithms) == 0:
-        print("Error: An algorithm (LOG_REG, SVM, XGB) must be requested per run.", file=sys.stderr)
+        print("Error: At least one algorithm (LOG_REG, SVM, XGB) must be requested per run.", file=sys.stderr)
```

Comment:
Aren't we still a max of one algorithm per run? If so, the language "at least one" could be misunderstood. For completeness we may have to add SGD in the parentheses or just remove all of the abbreviations.